auction-server/src/auction/service/verification.rs

```diff
@@ -578,25 +578,43 @@ impl Service<Svm> {
     }
 
     pub async fn simulate_bid(&self, bid: &entities::BidCreate<Svm>) -> Result<(), RestError> {
```

Comment:
I think calling this function recursively will make the code more readable (instead of having loop)

Reply:
I also considered that but that was also a bit weird. Refactored a bit to make it more readable